pos_mail_receipt/static/src/js/screens.js

```diff
@@ -0,0 +1,99 @@
+            } else {
+                this.gui.show_popup('textinput', {
+                    'title':_t('E-mail address to use'),
+                    'value': '',
+                    'confirm': function(value) {
+                        self._send_email_server(self.pos.get_order().name, {"email": value, "body_from_ui": body_from_ui});
+                    }
+                });
+            }
+        },
+        // ask the server to send the ticket as e-mail
+        // available options:
+        // - timeout: timeout for the rpc call in ms
+        // - email: email to use instead of the partner's
+        // returns a deferred
+        _send_email_server: async function (order, options) {
+            var self = this;
+            options = options || {};
+            var timeout = typeof options.timeout === 'number' ? options.timeout : 7500;
+            this.$('.button.email').addClass("highlight");
+            while (self.pos.get("synch").state == "connecting") {
+                await sleep(1000);
+            }
+
+            return rpc.query({
+                    model: 'pos.order',
+                    method: 'send_mail_receipt',
+                    args: [order, options["email"], options["body_from_ui"]],
+                }, {
+                    timeout: timeout,
+                })
+                .then(function (result) {
+                    return true
+                }).fail(function (type, error){
+                    var connection_problem = true;
+                    for (var i = 0; i < self.pos.db.get_orders().length; i++) {
+                        if (order == self.pos.db.get_orders()[i].data.name) {
+                            self.pos.db.get_orders()[i].data["email"] = options["email"] || false;
+                            connection_problem = false;
+                        }
+                    }
+                    if(error.code === 200 ){
+                        // Business Logic Error, not a connection problem
+                        //if warning do not need to display traceback!!
+                        if (error.data.exception_type == 'warning') {
+                            delete error.data.debug;
+                        }
+
+                        self.gui.show_popup('error-traceback',{
+                            'title': error.data.message,
+                            'body':  error.data.debug
+                        });
+                         this.$('.button.email').removeClass("highlight");
+                    }
+                    if(connection_problem){
+                        self.gui.show_popup('error',{
+                            'title': _t('The e-mail could not be sent'),
+                            'body': _t('Check your internet connection and try again.'),
+                        });
+                         this.$('.button.email').removeClass("highlight");
```

Comment:
I'm testing this branch and getting an error when the email can't be sent, stating `this.$ is not a function` referring to this line. I think this `this` should be a `self` created on line 46, as the this is lost here. Same goes for the `this` on line 82. This replacement fixed it for me.